src/ica_api.c

```diff
@@ -1745,7 +1756,7 @@ int ica_x25519_key_set(ICA_X25519_CTX *ctx,
 	UNUSED(pub);
 	return EPERM;
 #else
-	if (check_fips() || !msa9_switch || ctx == NULL)
+	if (check_fips(NID_X25519) || !msa9_switch || ctx == NULL)
 		return -1;
```

Comment:
I always wondered why the Edwards functions in most cases return -1 instead of a useful errno value, like the other functions do.

Reply:
Yes. I could change these returns to something like EPERM or EIO. But then we would have to distinguish between "no msa9", which would be EPERM or EIO, and ctx == NULL, which would be more like EINVAL. Unfortunately we externally document the rc = -1 in ica_api.h and the libica book. So I would do that later with a cleanup patch. It's noted on my todo list ;-)

Reply:
... and this might be considered as an API breakage ...